Analyzer/src/AnalyzeLepTrigger.cc

```diff
@@ -109,356 +97,158 @@ void AnalyzeLepTrigger::Loop(NTupleReader& tr, double weight, int maxevents, boo
+                        { "el_pt40_trig_5jCut",   passBaseline && foundMuonPt40 && passMuonTriggers && pass5JetCut }, 
+                        { "el_pt40_trig_6jCut",   passBaseline && foundMuonPt40 && passMuonTriggers && pass6JetCut }, 
+                        
+                        { "el_pt40_noTrig_1jCut", passBaseline && foundMuonPt40 && pass1JetCut }, 
+                        { "el_pt40_noTrig_2jCut", passBaseline && foundMuonPt40 && pass2JetCut }, 
+                        { "el_pt40_noTrig_3jCut", passBaseline && foundMuonPt40 && pass3JetCut }, 
+                        { "el_pt40_noTrig_4jCut", passBaseline && foundMuonPt40 && pass4JetCut }, 
+                        { "el_pt40_noTrig_5jCut", passBaseline && foundMuonPt40 && pass5JetCut }, 
+                        { "el_pt40_noTrig_6jCut", passBaseline && foundMuonPt40 && pass6JetCut } 
+                    };
     
-                        if( passElectronTriggers ) {
-                            my_histos["h_trig_num_"+kv.first+"_wLepPtBin"]->Fill( Electrons.at( myGoodElectronIndex ).Pt(), eventweight );
-                            my_histos["h_trig_num_"+kv.first+"_wLepEtaBin"]->Fill( Electrons.at( myGoodElectronIndex ).Eta(), eventweight );
-                            my_2d_histos["h2_trig_num_"+kv.first+"_wLepPtLepEtaBin"]->Fill( Electrons.at( myGoodElectronIndex ).Pt(), Electrons.at( myGoodElectronIndex ).Eta(), eventweight );
-                        }
-                    }
+                    fillHistos(cut_map_elTriggers, passElectronTriggers, Electrons.at( myGoodElectronIndex ), eventweight);
                 }
             }
         }
         
-        //------------------------------------------------------
-        //-- Do the Muon Trigger Efficiency on the Single Electron Dataset
-        //------------------------------------------------------
+        //-----------------------------------------------------------------------
+        //-- Do the Muon Trigger Efficiency on the Single Electron Dataset and MC
+        //-----------------------------------------------------------------------
         
-        if( filetag == "2016_Data_SingleElectron" || filetag == "2017_Data_SingleElectron" ) {
-            
-            if( !passOfflineBaseline  || !atLeastOneGoodEl ) continue;
-
-            bool foundElectronPt40      = false;
-            
-            //Require a good electron in the single electron dataset
-            for( unsigned int itEl = 0; itEl < Electrons.size(); ++itEl ) {
-                if( !GoodElectrons.at( itEl ) ) continue; 
+        if( (filetag.find("SingleElectron") != std::string::npos || filetag.find("EGamma") != std::string::npos || runtype == "MC") ) {
 
-                TLorentzVector myElectron = Electrons.at( itEl );
+            bool atLeastOneGoodEl = ( std::any_of( GoodElectrons.begin(), GoodElectrons.end(), [] ( bool boolEl ) { return boolEl; } ) );
```

Comment:
again don't loop over electrons when you can just use the NGoodElectron variable